src/app/shell/personal-cabinet/provider/provider-workshops/provider-workshops.component.html

```diff
@@ -1,16 +1,27 @@
 <div class="add-wrapper">
   <p class="text">Тут ви можете створити гурток, секцію або групу для навчання</p>
-  <a [routerLink]="['/create-workshop', ModeConstants.NEW]"><button class="btn" mat-raised-button>Додати гурток</button></a>
+  <a [routerLink]="['/create-workshop', ModeConstants.NEW]"
+    ><button class="btn" mat-raised-button>Додати гурток</button></a
+  >
 </div>
-<ng-container *ngIf="!(isLoadingCabinet$ | async)">
-  <div class="card-wrapper">
+
+<ng-container *ngIf="!(isLoadingCabinet$ | async) && !!workshops">
+  <div class="card-wrapper" *ngIf="workshops?.totalAmount; else isEmptyList">
     <app-workshop-card
       class="card"
-      *ngFor="let workshop of workshops$ | async"
+      *ngFor="let workshop of workshops.entities"
```

Comment:
pls provide trackBy for better performance

Reply:
in this case, it is not relevant, because we won't change a single item, we will receive the whole list dynamically, so any way we will rerender the whole list